app/services/holdings_request_service.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class HoldingsRequestService
+  attr_reader :args, :locations, :libraries, :exclude
+
+  def initialize(args)
+    @args = args
+    @libraries_data = libraries_data
+    @locations = libraries_data['locations']
+    @libraries = libraries_data['libraries']
+    @exclude = libraries_data['exclude_from_email_availability']
+  end
+
+  def make_holdings_request
+    sirsi_uri = URI(Settings.sirsi_url + args)
+
+    Net::HTTP.get(sirsi_uri)
+  end
+
+  def holdings_availability
+    holdings = Hash.from_xml(make_holdings_request)
+    availability = get_location_data(holdings)
+    if availability.present?
+      availability_to_string(availability)
+    end
+  end
+
+  private
+
+    def get_location_data(holdings)
+      call_info = holdings['LookupTitleInfoResponse']['TitleInfo']['CallInfo']
+
+      if call_info.is_a?(Array)
+        get_data_from_call_info(call_info)
+      else
+        library_data = get_library_data(call_info)
+        return [] if call_info['libraryID'] == 'ONLINE' || library_data.blank?
```

Comment:
your refactor looks great, this guard clause can be pushed down to `get_library_data` method if you like.